hrjobcontract/CRM/Hrjobcontract/Import/Form/Summary.php

```diff
@@ -1,4 +1,12 @@
 <?php
 class CRM_Hrjobcontract_Import_Form_Summary extends CRM_Hrjobcontract_Import_Form_SummaryBaseClass {
-  protected $_parser = 'CRM_Hrjobcontract_Import_Parser_Api';
-}
\ No newline at end of file
+
+  protected function getParser() {
```

Comment:
I can see 2 problems with this `getParser()` method:

1. In all the places where this method is called, you instantiate the parser right after the call. Something like:
```php
$parser = $this->getParser();
$parser = new $parser($some, $params);
``` 
If the whole idea is to build a parser instance, wouldn't it make more sense for the `getParser()` to return that instead of the class name?

2. This `getParser()` method is duplicated in three different classes and it would be better to avoid that. Here are some suggestions:
- Create a `ParserFactory` class, with a method like `create($inputMode, $other, $params, $to, $constructor)`, which will return an instance of the correct parser for the given inputMode
- Move this method to a trait and use it in these classes